microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/java/validators/annotations/AnnotationRulesJavaASTValidator.java

```diff
@@ -99,9 +102,16 @@ private void validateAnnotationAttributeValue(Expression attributeValueExpr, Ann
 
 		// Get the value of the AST attribute
 		String valueAsString = attributeValueExpr.toString();
-		if (StringUtils.isEmpty(valueAsString)) {
+
+		// Ensure value of AST attribute is a valid integer or an expression that can be evaluated
+		if (StringUtils.isEmpty(valueAsString)
+				|| (!isInteger(attributeValueExpr) && !isInfixIntegerExpression(attributeValueExpr))) {
 			return;
 		}
+
+		if (attributeValueExpr instanceof InfixExpression) {
+			valueAsString = attributeValueExpr.resolveConstantExpressionValue().toString();
+		}

```

Comment:
I would make the following change :

```diff
diff --git a/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/java/validators/annotations/AnnotationRulesJavaASTValidator.java b/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/java/validators/annotations/AnnotationRulesJavaASTValidator.java
index 84d823c..df13bef 100644
--- a/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/java/validators/annotations/AnnotationRulesJavaASTValidator.java
+++ b/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/java/validators/annotations/AnnotationRulesJavaASTValidator.java
@@ -100,18 +100,18 @@ public class AnnotationRulesJavaASTValidator extends JavaASTValidator {
 			return;
 		}
 
-		// Get the value of the AST attribute
-		String valueAsString = attributeValueExpr.toString();
-
 		// Ensure value of AST attribute is a valid integer or an expression that can be evaluated
-		if (StringUtils.isEmpty(valueAsString)
-				|| (!isInteger(attributeValueExpr) && !isInfixIntegerExpression(attributeValueExpr))) {
+		if (!isInteger(attributeValueExpr) && !isInfixIntegerExpression(attributeValueExpr)) {
 			return;
 		}
 
-		if (attributeValueExpr instanceof InfixExpression) {
-			valueAsString = attributeValueExpr.resolveConstantExpressionValue().toString();
+		// Get the value of the AST attribute
+		String valueAsString = attributeValueExpr.resolveConstantExpressionValue().toString();
+
+		if (StringUtils.isEmpty(valueAsString)) {
+			return;
 		}
+
 		// Validate the value with the rule
 		String validationResult = JavaASTValidatorRegistry.getInstance().validate(valueAsString, attributeRule);
 		if (validationResult != null) {
```

`resolveConstantExpressionValue()` ca be called even when the expression isn't an infix expression. We can place the check for `isEmpty` after that. Ultimately it's better to avoid calling `toString()` if we can.